### PR TITLE
Add Storage extension methods

### LINKED ISSUE #6983: Support loading of custom credential types from configuration in ASP.NET Core integration
For storage.

### PR DESCRIPTION
**Pull request by pakrym**
Add support for loading composite objects from the configuration. Fixes https://github.com/Azure/azure-sdk-for-net/issues/6983.

Use longest constructor possible when loading from the configuration.

### CODE REVIEW

File: sdk/core/Azure.Core.Extensions/samples/Startup.cs [hunk 1]
```diff
@@ -48,11 +50,14 @@ public void ConfigureServices(IServiceCollection services)
 
                 // Advanced configure global defaults
                 builder.ConfigureDefaults((options, provider) =>  options.AddPolicy(HttpPipelinePosition.PerCall, provider.GetService<DependencyInjectionEnabledPolicy>()));
+
+                builder.AddBlobServiceClient(Configuration.GetSection("Storage"))
```
Comment (tg-msft) [inline sdk/core/Azure.Core.Extensions/samples/Startup.cs]: Need to add the updated config file?

**Review by tg-msft — APPROVED** [review]

File: sdk/core/Azure.Core.Extensions/samples/Startup.cs (hunk 1 above)
Comment (loic-sharma) [reply sdk/core/Azure.Core.Extensions/samples/Startup.cs]: Will we need to pass the connection string to the `AddBlobServiceClient` call? Or can that be part of the `Storage` configuration section?

Comment (loic-sharma) [reply sdk/core/Azure.Core.Extensions/samples/Startup.cs]: Will we need to pass the connection string to the `AddBlobServiceClient` call? Or can that be part of the `Storage` configuration section?

Comment (pakrym) [reply sdk/core/Azure.Core.Extensions/samples/Startup.cs]: The connection string can be a part of the configuration.

**Pull request merged**